TestRunner.psm1

```diff
@@ -269,203 +276,379 @@ DEALLOCATE reorganizeOrRebuildCommands_cursor;
+        Write-InfoLog "Writing to $testResultsPath"
 
-        $currentTestResultsFolder = $Using:testResultsPath
+        $databaseInstance = Get-ConfigValue -Config $Config -Key "PERF_DB_SERVER"
+        $databaseServer = Get-DatabaseServerName $databaseInstance
 
+        $webServer = Get-ConfigValue -Config $Config -Key "PERF_WEB_SERVER"
+        $sqlBackupFile = ""
+        $restoreDatabase = $false
         try {
-            $csvPath = "$Using:thisFolder\$currentTestResultsFolder\$Using:testType.$server.csv"
+            $restoreDatabase = [System.Convert]::ToBoolean($(Get-ConfigValue -Config $Config -Key "PERF_DB_RESTORE"))
+        } catch {
+            throw "Invalid configuration value for PERF_DB_RESTORE, should be a boolean"
+        }
+        if ($restoreDatabase) {
+            $sqlBackupFile = Get-ConfigValue -Config $Config -Key "PERF_SQL_BACKUP_FILE"
+        }
 
-            if ($server -eq 'localhost') {
-                $actualPerformanceCounters = Compare-Object (typeperf -q) $expectedPerformanceCounters -PassThru -IncludeEqual -ExcludeDifferent
+        $url = Get-ConfigValue -Config $Config -Key "PERF_API_BASEURL"
+        $key = Get-ConfigValue -Config $Config -Key "PERF_API_KEY"
+        $secret = Get-ConfigValue -Config $Config -Key "PERF_API_SECRET"
+        $clientCount = Get-ConfigValue -Config $Config -Key "PERF_CONNECTION_LIMIT" -Optional
+        if ($null -eq $clientCount) { $clientCount = 5 }
+        $contentType = Get-ConfigValue -Config $Config -Key "PERF_CONTENT_TYPE" -Optional
+        $resourceList = Get-ConfigValue -Config $Config -Key "PERF_RESOURCE_LIST" -Optional
+        $pageSize = Get-ConfigValue -Config $Config -Key "PERF_API_PAGE_SIZE" -Optional
+        $logLevel = Get-ConfigValue -Config $Config -Key "PERF_LOG_LEVEL" -Optional
+        $description =Get-ConfigValue -Config $Config -Key "PERF_DESCRIPTION" -Optional
+
+        if ($restoreDatabase) {
+            Write-InfoLog "Restoring $databaseName from $sqlBackupFile"
+
+            if (Test-IsLocalhost $databaseServer) {
+                Reset-OdsDatabase -BackupFilename $sqlBackupFile -DatabaseName $databaseInstance
             }
             else {
-                $sessionOptions = New-PSSessionOption -SkipCACheck -SkipCNCheck
-                $psSession = New-PSSession -ComputerName $server -Credential $credential -UseSSL -SessionOption $sessionOptions -ErrorAction Stop
-
-                # Prepare the remote session with the contents of this file.
-                Invoke-Command -Session $psSession -ArgumentList @($commonFunctions, $Using:configJson) -ScriptBlock {
-                    param($commonFunctions, $configJson)
-                    Invoke-Expression $commonFunctions
-                    Set-GlobalsFromConfig $configJson
-                }
+                Invoke-RemoteCommand $databaseServer (Get-CredentialOrDefault $databaseServer) `
+                    -ArgumentList @($sqlBackupFile, $databaseInstance) -ScriptBlock {
```

Comment:
Same as above. $databaseInstance contains the value from PERF_DB_SERVER, while restore method is expecting database name

Reply:
After the fix, restore ran without error:

![image](https://user-images.githubusercontent.com/9324390/165563617-4a7e80d9-6ff6-4c0d-ad0a-d4f04aeb8a7d.png)